pkgs/racket-test-core/tests/racket/file.rktl

```diff
@@ -349,14 +349,123 @@
+(let-values ([(required accepted) (procedure-keywords make-temporary-directory)])
+  (test '() 'make-temporary-directory-no-required-keywords required)
+  (test '(#:base-dir) 'make-temporary-file-accepts-keywords accepted))
+
 (let ([tf (make-temporary-file)])
   (let-values ([(base name dir?) (split-path tf)])
-    (test #t 'make-temporary-file-uses-srcloc (and (regexp-match #rx"file.rktl" (path->bytes name)) #t)))
+    (test #t 'make-temporary-file-uses-srcloc (regexp-match? #rx"file.rktl" name)))
   (delete-file tf))
+(let ([tf (make-temporary-file #:copy-from 'directory)])
+  (let-values ([(base name dir?) (split-path tf)])
+    (test #t 'make-temporary-file-with-kw-uses-srcloc (regexp-match? #rx"file.rktl" name)))
+  (delete-directory tf))
+(let ([tf (make-temporary-directory)])
+  (let-values ([(base name dir?) (split-path tf)])
+    (test #t 'make-temporary-directory-uses-srcloc (regexp-match? #rx"file.rktl" name)))
+  (delete-directory tf))
+(let ([tf (make-temporary-directory #:base-dir (find-system-path 'temp-dir))])
+  (let-values ([(base name dir?) (split-path tf)])
+    (test #t 'make-temporary-directory-with-kw-uses-srcloc (regexp-match? #rx"file.rktl" name)))
+  (delete-directory tf))
+
+(let ([make-temporary-file ((λ (x) x) make-temporary-file)])
+  (define dir (make-temporary-file "mtf-ho-dir~a" 'directory))
+  (test #t 'make-temporary-file-in-ho-position (directory-exists? dir))
+  (define tf (make-temporary-file #:base-dir dir))
+  (test #t 'make-temporary-file-in-ho-position-with-kw (file-exists? tf))
+  (delete-file tf)
+  (delete-directory dir))
+(let ([make-temporary-directory ((λ (x) x) make-temporary-directory)])
+  (define dir (make-temporary-directory "mtd-ho-dir~a"))
+  (test #t 'make-temporary-directory-in-ho-position (directory-exists? dir))
+  (define td (make-temporary-directory #:base-dir dir))
+  (test #t 'make-temporary-file-in-ho-position-with-kw (directory-exists? td))
+  (delete-directory td)
+  (delete-directory dir))
 
-(let ([tf ((λ (t) (t)) make-temporary-file)])
-  (test #t 'make-temporary-file-in-ho-position (file-exists? tf))
-  (delete-file tf))
+(let ()
+  (define a (make-temporary-file))
+  (call-with-output-file a
+    #:exists 'truncate
+    (λ (out) (write 'copy-from out)))
+  (define b (make-temporary-file #:copy-from a))
+  (delete-file a)
+  (test 'copy-from 'make-temporary-file-copy-from (call-with-input-file b read))
+  (delete-file b))
+
+(define rx:tmp-file #rx"make-temporary-file")
+(define rx:tmp-dir #rx"make-temporary-directory")
+(err/rt-test (make-temporary-file "bad\0~a") exn:fail:contract? rx:tmp-file)
+(err/rt-test (make-temporary-directory "bad\0~a") exn:fail:contract? rx:tmp-dir)
+(err/rt-test (make-temporary-file "bad~x") exn:fail:contract? rx:tmp-file)
+(err/rt-test (make-temporary-directory "bad~x") exn:fail:contract? rx:tmp-dir)
+(let* ([temp-dir (find-system-path 'temp-dir)]
+       [absolute-template
+        (path->string (build-path temp-dir "absolute~a"))])
```

Comment:
This is the part that causes the error in the windows tests -- using a path as a string isn't guaranteed (and here doesn't) produce a usable format string, and might have other problems as well.